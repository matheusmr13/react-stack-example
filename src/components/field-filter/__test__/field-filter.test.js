import React from 'react';
import DatetimeFilter from 'components/field-filter/datetime-filter';
import UnlimitedIntegerFilter from 'components/field-filter/unlimited-integer-filter';
import LimitedIntegerFilter from 'components/field-filter/limited-integer-filter';
import ListFilter from 'components/field-filter/list-filter';
import FieldFilter from 'components/field-filter';
import {
	shallow
} from 'enzyme';
import moment from 'moment';

const filters = {
	filters: [{
		id: 'locale',
		name: 'Locale',
		values: [{
			value: 'en_AU',
			name: 'en_AU'
		}, {
			value: 'de_DE',
			name: 'de_DE '
		}, {
			value: 'pt_BR',
			name: 'pt_BR'
		}, {
			value: 'fr_FR',
			name: 'fr_FR'
		}, {
			value: 'en_US',
			name: 'en_US'
		}, {
			value: 'es_AR',
			name: 'es_AR'
		}]
	}, {
		id: 'country',
		name: 'País',
		values: [{
			value: 'AU',
			name: 'Australia'
		}, {
			value: 'DE',
			name: 'Alemanhã'
		}, {
			value: 'BR',
			name: 'Brasil'
		}, {
			value: 'PT',
			name: 'Portugal'
		}, {
			value: 'en_US',
			name: 'EUA'
		}, {
			value: 'RU',
			name: 'Rússia'
		}]
	}, {
		id: 'timestamp',
		name: 'Data e Horário',
		validation: {
			primitiveType: 'STRING',
			entityType: 'DATE_TIME',
			pattern: 'yyyy-MM-ddTHH:mm:ss'
		}
	}, {
		id: 'limit',
		name: 'Quantidade',
		validation: {
			primitiveType: 'INTEGER',
			min: 1,
			max: 50
		}
	}, {
		id: 'offset',
		name: 'Página',
		validation: {
			primitiveType: 'INTEGER'
		}
	}]
};

describe('DatetimeFilter component', () => {
			it('should render properly', () => {
				const datetimeFilter = shallow( < DatetimeFilter spec = {
						{
							name: 'Date field'
						}
					}
					/>);
					expect(datetimeFilter).toMatchSnapshot();
				}); it('should merge date and time', () => {
					let lastChangedDate;

					const onChange = (value) => {
						lastChangedDate = value;
					};
					const datetimeFilter = shallow( < DatetimeFilter spec = {
							{
								name: 'Date field'
							}
						}
						onChange = {
							onChange
						}
						/>);
						expect(lastChangedDate).toBeUndefined();

						const dateMoment = moment({
							years: 2018,
							month: 10,
							date: 15
						});
						const hourMoment = moment({
							minutes: 40,
							hours: 10
						}); datetimeFilter.instance().onChange('date', dateMoment.toDate()); datetimeFilter.instance().onChange('time', hourMoment.toDate());

						expect(lastChangedDate).not.toBeUndefined(); expect(lastChangedDate).toBe('2018-11-15T10:40:00-02:00');
					});
			});