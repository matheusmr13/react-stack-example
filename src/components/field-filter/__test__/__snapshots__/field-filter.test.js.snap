// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DatetimeFilter component should render properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DatetimeFilter
    spec={
        Object {
            "name": "Date field",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <DatePicker
          autoOk={false}
          container="dialog"
          disableYearSelection={false}
          disabled={false}
          firstDayOfWeek={1}
          hideCalendarDate={false}
          hintText="Date field"
          onChange={[Function]}
          openToYearSelection={false}
          style={Object {}}
/>,
        <TimePicker
          autoOk={false}
          cancelLabel="Cancel"
          defaultTime={null}
          disabled={false}
          format="ampm"
          hintText="Date field"
          minutesStep={1}
          okLabel="OK"
          onChange={[Function]}
          pedantic={false}
          style={Object {}}
          value={null}
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "autoOk": false,
          "container": "dialog",
          "disableYearSelection": false,
          "disabled": false,
          "firstDayOfWeek": 1,
          "hideCalendarDate": false,
          "hintText": "Date field",
          "onChange": [Function],
          "openToYearSelection": false,
          "style": Object {},
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "autoOk": false,
          "cancelLabel": "Cancel",
          "defaultTime": null,
          "disabled": false,
          "format": "ampm",
          "hintText": "Date field",
          "minutesStep": 1,
          "okLabel": "OK",
          "onChange": [Function],
          "pedantic": false,
          "style": Object {},
          "value": null,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <DatePicker
            autoOk={false}
            container="dialog"
            disableYearSelection={false}
            disabled={false}
            firstDayOfWeek={1}
            hideCalendarDate={false}
            hintText="Date field"
            onChange={[Function]}
            openToYearSelection={false}
            style={Object {}}
/>,
          <TimePicker
            autoOk={false}
            cancelLabel="Cancel"
            defaultTime={null}
            disabled={false}
            format="ampm"
            hintText="Date field"
            minutesStep={1}
            okLabel="OK"
            onChange={[Function]}
            pedantic={false}
            style={Object {}}
            value={null}
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "autoOk": false,
            "container": "dialog",
            "disableYearSelection": false,
            "disabled": false,
            "firstDayOfWeek": 1,
            "hideCalendarDate": false,
            "hintText": "Date field",
            "onChange": [Function],
            "openToYearSelection": false,
            "style": Object {},
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "autoOk": false,
            "cancelLabel": "Cancel",
            "defaultTime": null,
            "disabled": false,
            "format": "ampm",
            "hintText": "Date field",
            "minutesStep": 1,
            "okLabel": "OK",
            "onChange": [Function],
            "pedantic": false,
            "style": Object {},
            "value": null,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
